avro-serde/src/main/java/io/confluent/kafka/streams/serdes/SpecificAvroDeserializer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.streams.serdes;
+
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.Collections;
+import java.util.Map;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+
+public class SpecificAvroDeserializer<T extends org.apache.avro.specific.SpecificRecord>
+    implements Deserializer<T> {
+
+  private final KafkaAvroDeserializer inner;
+
+  /**
+   * Implementation detail: This constructor is used by Kafka's Streams API.
+   */
+  public SpecificAvroDeserializer() {
+    inner = new KafkaAvroDeserializer();
+  }
+
+  public SpecificAvroDeserializer(final SchemaRegistryClient client) {
+    this(client, Collections.<String, Object>emptyMap());
+  }
+
+  public SpecificAvroDeserializer(final SchemaRegistryClient client,
+                                  final Map<String, ?> deserializerConfig) {
+    inner = new KafkaAvroDeserializer(
+        client,
+        ConfigurationUtils.withSpecificAvroEnabled(deserializerConfig));
+  }
+
+  @Override
+  public void configure(final Map<String, ?> deserializerConfig,
+                        final boolean isDeserializerForRecordKeys) {
+    inner.configure(
+        ConfigurationUtils.withSpecificAvroEnabled(deserializerConfig),
+        isDeserializerForRecordKeys);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public T deserialize(final String topic, final byte[] bytes) {
+    return (T) inner.deserialize(topic, bytes);
```

Comment:
Same question here about the potential class cast error and wrapping it in an exception of a different type.